### PR TITLE
Feature security schema per method

### PR DESCRIPTION
**Pull request by Jonathan-Fan**
| Q                       | A   |
| ---                     | --- |
| Bug fix?                | no |
| New feature?            | yes |
| BC breaks?              | yes |
| Deprecations?           | no |
| Continuous improvement? | yes |
| Tests pass?             | yes |
| Changelog updated?      | yes |
| Fixed issue           | [Issue-115](https://github.com/temando/open-api-renderer/issues/115) |

Features:

 * Visualise per method the security schemes required

### CODE REVIEW

File: test/components/__snapshots__/SecurityContainer.test.js.snap [hunk 1]
```diff
@@ -5,17 +5,9 @@ exports[`<SecurityContainer /> renders apiKey type 1`] = `
...
-     
-    <code
-      className={undefined}
-    >
-      type=
-      apiKey
-    </code>
-  </h3>
-  <div>
+  <div
```
Comment (brendo) [inline test/components/__snapshots__/SecurityContainer.test.js.snap]: Could you please update these tests with the `placedIn` prop so the snapshot doesn't change?

File: src/components/SecurityContainer/SecurityContainer.js [hunk 2]
```diff
@@ -103,5 +118,6 @@ export default class SecurityContainer extends PureComponent {
 SecurityContainer.propTypes = {
   id: PropTypes.string,
   security: PropTypes.object,
-  classes: PropTypes.object
+  classes: PropTypes.object,
+  placedIn: PropTypes.string
```
Comment (brendo) [inline src/components/SecurityContainer/SecurityContainer.js]: Consider making this propType an enum with the two values we know about. Will help others understand the codebase faster.

**Review by brendo — CHANGES_REQUESTED** [review]
Thanks @Jonathan-Fan! This is mostly pretty good. I previewed it using the Petstore schema though and noticed that it's now quite "heavy" and repetitive.  

What do you think about having an Authentication header and name of the security scheme which links back up to the description at the top of the page?

File: src/components/SecurityContainer/SecurityContainer.js (hunk 2 above)
Comment (Jonathan-Fan) [reply src/components/SecurityContainer/SecurityContainer.js]: Changed to `oneOf`.

File: test/components/__snapshots__/SecurityContainer.test.js.snap (hunk 1 above)
Comment (Jonathan-Fan) [reply test/components/__snapshots__/SecurityContainer.test.js.snap]: Added `placedIn`, but snapshot still changes since introduce `simple` className.

**Comment by Jonathan-Fan** [comment]
Regarding to `Authentication Schema`, I think it is a little bit annoying when you check a method and have to jump to the top check the Authentication and then jump back. So in my opinion, we can show/hide the `Authentication Section` for each method.

**Comment by brendo** [comment]
That too would also be fine. By default it'd be collapsed?

Authentication: authName

**Comment by Szasza** [comment]
@Jonathan-Fan thank you for getting the implementation this far.

Would it be possible to make the changes @brendo has asked and get this feature merged in?

Thank you in advance for the response.

**Comment by Jonathan-Fan** [comment]
Hi @Szasza , 
So busy these days,  will do it during Xmas time. 

Regards
Jonathan

**Comment by Szasza** [comment]
Hi @Jonathan-Fan ,

Thank you for the response, it is much appreciated :)

Kind regards,
Szasza

**Review by brendo — APPROVED** [review]

**Comment by Szasza** [comment]
:fireworks:

**Comment by brendo** [comment]
This is now deployed as part of `v0.4.0`

**Pull request merged**